src/sql/parts/notebook/cellViews/textCell.component.ts

```diff
@@ -149,6 +151,24 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		return content;
 	}
 
+	// Only replace vscode-resource with file when in the same (or a sub) directory
+	// This matches Jupyter Notebook viewer behavior
+	private convertVsCodeResourceToFileInSubDirectories(htmlContent: string): string {
+		let htmlContentCopy = htmlContent;
+		while (htmlContentCopy.search('(?<=img src=\"vscode-resource:)') > 0) {
+			let pathStartIndex = htmlContentCopy.search('(?<=img src=\"vscode-resource:)');
+			let pathEndIndex = htmlContentCopy.slice(pathStartIndex).indexOf('\" ') + pathStartIndex;
+			let filePath = htmlContentCopy.substring(pathStartIndex, pathEndIndex);
+			// If the asset is in the same folder or a subfolder, replace 'vscode-resource:' with 'file:', so the image is visible
+			if (!path.relative(path.dirname(this.cellModel.notebookModel.notebookUri.fsPath), filePath).includes('..')) {
```

Comment:
For relative paths how will this `path.relative` end up looking? For example:
```
./its/a/subdir/with/../some/../in/the/name
../actualRelativeDir
```

Reply:
../actualRelativeDir